### PR TITLE
ota-architecture: add lmp-device security diagram

### PR DESCRIPTION
**Pull request by ldts**
## Readiness

* [ ] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

_Why merge this PR? What does it solve?_

## Checklist

_Optional. Add a 'x' to steps taken._
_You can fill this out after opening the PR. "Did I..."_

* [ ] Run spelling and grammar check, preferably with linter.
* [ ] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [ ] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [ ] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [ ] View HTML in a browser to check rendering.
* [ ] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [ ] follow best practices for commits.
  * [ ] Descriptive title written in the imperative.
  * [ ] Include brief overview of QA steps taken.
  * [ ] Mention any related issues numbers.
  * [ ] End message with sign off/DCO line (`-s, --signoff`).
  * [ ] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [ ] Squash commits if needed.
* [ ] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 07600b20d9e18f30761a2dd419c7fd54326bb8be are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1981/docs/artifacts/html/index.html

**Comment by ldts** [comment]
changed this page with the device security info:
https://ci.foundries.io/projects/fio-docs/builds/1980/docs/artifacts/html/reference-manual/ota/ota-architecture.html

**Comment by doanac** [comment]
Docs for f6d3e5dbe33a059464107ad8ae8ff256f18e2e46 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1983/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for e4465152303f8023375dd7d1c2595a8ae7bd249a are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1985/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for b013d600115f6dba6af36d8bf9dec55bd5f67dfc are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1987/docs/artifacts/html/index.html

**Comment by doanac** [comment]
Docs for 9a793424995da895ffc899db1d7b792bce604e54 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1988/docs/artifacts/html/index.html

File: source/reference-manual/ota/ota-architecture.rst [hunk 1]
```diff
@@ -8,7 +8,10 @@ Architecture Overview
 At a high level the system consists of three entities:
 
  * LmP Devices
-   - running aktualizr-lite and fioconfig
+   
+   - Execute lmp-device-register to associate the device with the gateway
```
Comment (kprosise) [inline source/reference-manual/ota/ota-architecture.rst:12]:
```suggestion
   - Execute ``lmp-device-register`` to associate the device with the gateway
```

File: source/reference-manual/ota/ota-architecture.rst [hunk 2]
```diff
@@ -8,7 +8,10 @@ Architecture Overview
...
+     
+   - Run aktualizr-lite and fioconfig daemons.
```
Comment (kprosise) [inline source/reference-manual/ota/ota-architecture.rst:14]:
```suggestion
   - Run ``aktualizr-lite`` and ``fioconfig`` daemons.
```

File: source/reference-manual/ota/ota-architecture.rst [hunk 3]
```diff
@@ -31,6 +34,24 @@ device gateway with HTTP GET requests on configurable intervals.
 Due to the fact devices are polling the server, REST API changes requested by
 tooling like fioctl happen asynchronously.
 
+How A Device Uses Security Hardware
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An LmP device uses Hardware Security Modules, TPM devices or
```
Comment (kprosise) [inline source/reference-manual/ota/ota-architecture.rst:40]:
```suggestion
An LmP device uses Hardware Security Modules, Trusted Platform Module (TPM) devices or
```

File: source/reference-manual/ota/ota-architecture.rst [hunk 4]
```diff
@@ -31,6 +34,24 @@ device gateway with HTTP GET requests on configurable intervals.
...
+Trusted Execution Environments via the PKCS#11 API.
```
Comment (kprosise) [inline source/reference-manual/ota/ota-architecture.rst:41]:
```suggestion
Trusted Execution Environments via the Public-Key Cryptography Standards #11 (PKCS#11) API.
```

**Review by kprosise — COMMENTED** [review]
Suggested changes to address linter findings, otherwise it looks good to me. However I lack the expertise to evaluate the diagram, but I recall it being discussed over slack, so from my end everything looks good once the comments have been resolved.

**Comment by doanac** [comment]
Docs for e072265b8b10f77180c1e4c596590a5b7da4a873 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/1990/docs/artifacts/html/index.html

**Comment by ldts** [comment]
> Suggested changes to address linter findings, otherwise it looks good to me. However I lack the expertise to evaluate the diagram, but I recall it being discussed over slack, so from my end everything looks good once the comments have been resolved.

thanks @kprosise, changed applied

**Review by kprosise — APPROVED** [review]
LGTM

**Pull request merged**